src/System.Linq/tests/ExceptTests.cs

```diff
@@ -125,5 +125,22 @@ public void ForcedToEnumeratorDoesntEnumerate()
             var en = iterator as IEnumerator<int>;
             Assert.False(en != null && en.MoveNext());
         }
+
+        [Fact]
+        public void HashSetWithBuiltInComparer_HashSetContainsNotUsed()
+        {
+            IEnumerable<string> input1 = new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "a" };
+            IEnumerable<string> input2 = new[] { "A" };
```

Comment:
This should also be a `HashSet`. If anything this would be the more likely to have an impact with a bogus optimisation as the implementation produces a set from `second`, then filters `first` based on it.
(That wouldn't even be a bad optimisation if we checked the comparer was correct first, the question is whether it's a common enough case to justify the check on whether that path could be taken).